mmdet3d/models/detectors/imvotenet.py

```diff
@@ -0,0 +1,759 @@
+                 pts_bbox_heads=None,
+                 pts_neck=None,
+                 img_backbone=None,
+                 img_neck=None,
+                 img_roi_head=None,
+                 img_rpn_head=None,
+                 img_mlp=None,
+                 fusion_layer=None,
+                 num_sampled_seed=None,
+                 train_cfg=None,
+                 test_cfg=None,
+                 pretrained=None):
+
+        super(ImVoteNet, self).__init__()
+
+        # point branch
+        if pts_backbone is not None:
+            self.pts_backbone = builder.build_backbone(pts_backbone)
+        if pts_neck is not None:
+            self.pts_neck = builder.build_neck(pts_neck)
+        if pts_bbox_heads is not None:
+            pts_bbox_head_common = pts_bbox_heads.common
+            pts_bbox_head_common.update(
+                train_cfg=train_cfg.pts if train_cfg is not None else None)
+            pts_bbox_head_common.update(test_cfg=test_cfg.pts)
+            pts_bbox_head_joint = pts_bbox_head_common.copy()
+            pts_bbox_head_joint.update(pts_bbox_heads.joint)
+            pts_bbox_head_pts = pts_bbox_head_common.copy()
+            pts_bbox_head_pts.update(pts_bbox_heads.pts)
+            pts_bbox_head_img = pts_bbox_head_common.copy()
+            pts_bbox_head_img.update(pts_bbox_heads.img)
+
+            self.pts_bbox_head_joint = builder.build_head(pts_bbox_head_joint)
+            self.pts_bbox_head_pts = builder.build_head(pts_bbox_head_pts)
+            self.pts_bbox_head_img = builder.build_head(pts_bbox_head_img)
+            self.pts_bbox_heads = [
+                self.pts_bbox_head_joint, self.pts_bbox_head_pts,
+                self.pts_bbox_head_img
+            ]
+            self.loss_weights = pts_bbox_heads.loss_weights
+
+        # image branch
+        if img_backbone:
+            self.img_backbone = builder.build_backbone(img_backbone)
+        if img_neck is not None:
+            self.img_neck = builder.build_neck(img_neck)
+        if img_rpn_head is not None:
+            rpn_train_cfg = train_cfg.img_rpn if train_cfg \
+                is not None else None
+            img_rpn_head_ = img_rpn_head.copy()
+            img_rpn_head_.update(
+                train_cfg=rpn_train_cfg, test_cfg=test_cfg.img_rpn)
+            self.img_rpn_head = builder.build_head(img_rpn_head_)
+        if img_roi_head is not None:
+            rcnn_train_cfg = train_cfg.img_rcnn if train_cfg \
+                is not None else None
+            img_roi_head.update(
+                train_cfg=rcnn_train_cfg, test_cfg=test_cfg.img_rcnn)
+            self.img_roi_head = builder.build_head(img_roi_head)
+
+        # fusion
+        if fusion_layer is not None:
+            self.fusion_layer = builder.build_fusion_layer(fusion_layer)
+            self.max_imvote_per_pixel = fusion_layer.max_imvote_per_pixel
+
+            # fusion layer exists -> stage 2 training -> freeze img branch
```

Comment:
1. May implement a function like `freeze_img_detector`to freeze layers
2. Add an explicit argument to do that. Should not use `fusion_layer` to determine whether image detector should be freezed.